index.js

```diff
@@ -652,4 +657,16 @@ HtmlWebpackPlugin.prototype.applyPluginsAsyncWaterfall = function (compilation)
   };
 };
 
+/**
+ * Get stats from compilation
+ */
+HtmlWebpackPlugin.prototype.getCompilationStats = function (compilation) {
+  var stats = compilationHashToStatsMap[compilation.hash];
+  if (!stats) {
+    stats = compilation.getStats().toJson();
```

Comment:
This `toJson` call can be passed a number of `false` options for specific keys to save more time, and limit generation to only the chunk information (which is all that's used by this plugin). You can see an example of that here in another perf PR: https://github.com/jantimon/html-webpack-plugin/pull/825/files#diff-168726dbe96b3ce427e7fedce31bb0bcL67